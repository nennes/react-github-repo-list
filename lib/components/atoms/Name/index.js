/* @flow */

import React from "react";
import { Repo as RepoType } from "../../../types";

const Name = ({ name, url }: RepoType) => {
  return (
    <div className="repo__name">
      <a href={url} target="_blank">
        <span title={name}>
          {name}
        </span>
      </a>
    </div>
  );
};

export default Name;
