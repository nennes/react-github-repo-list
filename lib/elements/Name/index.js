import React from "react";
import PropTypes from "prop-types";

const Name = ({ name, url }) => {
  return (
    <a className="repo-name" href={url} target="_blank">
      <span title={name}>
        {name}
      </span>
    </a>
  );
};

Name.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Name;
