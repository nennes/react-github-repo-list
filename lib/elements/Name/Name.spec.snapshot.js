/* global test, expect */

import React from "react";
import Name from "./index";
import renderer from "react-test-renderer";

test("Name element renders correctly", () => {
  const props = {
    name: "react-component-library",
    url: "http://www.test.com"
  };
  const element = renderer.create(<Name {...props} />);
  expect(element.toJSON()).toMatchSnapshot();
});
