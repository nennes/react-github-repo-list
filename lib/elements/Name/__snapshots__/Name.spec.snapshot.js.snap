// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name element renders correctly 1`] = `
<a
  className="repo-name"
  href="http://www.test.com"
  target="_blank"
>
  <span
    title="react-component-library"
  >
    react-component-library
  </span>
</a>
`;
